### ISSUE TITLE
Enhancement Request: Command to have flagship hold position (that is compatable with afterburners).

### ISSUE THREAD

**Issue by marphod** [endless-sky/endless-sky#2815]
... and return  to that position if moved.  i know you can do i manually with thrusters, but inevitably, i'm off by a little.  plus, if a hostile attacks or something explodes nearby, the flagship starts to drift.

Alternatively, a command to 'orbit' (like the behavior of trying to land on a restricted/hostile planet) would work.

**Comment by Amazinite** [comment]
Duplicate: #1706

**Comment by Bladewood** [comment]
I never really was satisfied by how LocalGod had decided to handle it.
I don't want to rely on an outfit to be able to stand still. If the AI can do it, why can't I?

**Comment by marphod** [comment]
... That's a terrible implementation.

It breaks afterburners, needs to be purchased, either requires weapon mass or breaks game consistency, should require a hardpoint or breaks realism (that reverse thrusters don't use a hardpoint continues to confuse me), and is the very definition of a 'hack'.

Why is adding new keybindings hard/avoided?  The code to do them has to exist, and the code to stop a ship also already exists. Seems like a 20-minutes-(plus-buildtime) change.

**Comment by Bladewood** [comment]
The problem is that there aren't enough.

These people are planning on porting Endless Sky to console (eventually).

**Comment by Rob59er** [comment]
The amount of money I would pay for this feature...
But in all seriousness, we now have more reason then ever to add it due to the sped up game time.

**Comment by tehhowch** [comment]
I'm curious: why would you ever need your flagship to sit perfectly still?

PS if the answer is "to farm combat rating automatically / without paying any attention", you might as well close this because it's going to get `unlikely`. Just fly outside the boundary and tell your escort fleet to hold position in the center of the system.
If you want them to keep returning to their held position, go support my existing PR that transparently converts between MOVE_TO and HOLD_POSITION orders depending on distance from the desired point ( #2591 ).

**Comment by Amazinite** [comment]
>But in all seriousness, we now have more reason then ever to add it due to the sped up game time.

I wouldn't use the 3x game speed option as a reason for needing a flagship hold position button. You can just turn it off then try to maneuver your ship to a stop, then speed time back up.

I don't know if it's just because I've gotten use to it, but I'm able to get my ship to an almost complete stand still within a minute or two as long as I'm not getting hit.

**Comment by ghoulavenger** [comment]
There are already solutions for farming combat rating automatically without paying attention, use a cloaking device with a few catalytic ramscoops (or wanderer perhaps, 4-5 wanderers are probably enough), or let your flagship drift while the rest of your fleet is holding position near the center of the system (ships always spawn near the center of the system so if you're drifting far away from the center, they'll never aggro you anyway -- it'll just take some time to get back to the center of the map).

So if that is the reason to oppose a hold position for the flagship I'm entirely unimpressed. As for practical purposes, having the hold position button would be nice for the Sad Archie mission, since you need to stay close to the star and be relatively still for a few minutes to trigger it.

At least the controller portability issue which I consider more relevant, it makes controls harder to map -- like you might need to hold two buttons at the same time or something. But perhaps some controls are antiquated or not useful anyway. Like the buy x5, x20, x500 button. To some degree, you may want to buy 5 or 20 outfits at a time if you're buying hand to hand weapons or the new remnant keystone mission, but you can just hold down the install button for the same effect. Practically speaking you will never use them to buy or sell cargo on the trading page when a buy and sell all button exists (you may be able to haul multiple cargo types but there is very little incentive to not haul the most profitable).

**Comment by gunqqer** [comment]
Quick question which one is the Sad Archie mission. I'm all for this change (I want it myself very much) but I don't remember doing that one. I probably have but idk

**Comment by ghoulavenger** [comment]
Sad Archie is a mission that is unmarked, you acquire it by doing the TMBR chain to its completion, which requires doing the Deep Archaeology chain. One of the guys from the band tells you the story of Sad Archie (the song) hence the name of the mission. It requires you visit a certain starsystem and pretty much emulate his story. Which requires powering down everything and meditating for awhile.

**Comment by gunqqer** [comment]
Ah that one, forgot about the TMBR chain. I may have to go find it and do it

**Comment by ghoulavenger** [comment]
You know, if this has to be handled as an outfit, it doesn't necessarily need to be like afterburners. The reasoning behind why ships don't stop on their own is because there is no opposing force such as friction. So you need to offer a counter force of your own, such as reverse thrusters. But reverse thrusters don't perfectly handle stopping your ship because you can overextend your thrusters, causing backward acceleration.

Now, if people want to drive in reverse I don't have anything against that, so we need to leave that functionality in. But we do have a couple options regarding improving the function of reverse thrusters. We can offer an outfit which stops you from driving in reverse. Basically, altering the function of reverse thrusters to function as brakes. If you don't like the outfit the other alternative is a simple option in the settings (either one preserves the initial use of reverse thrusters while making it easier to stop). But this does not address drift. Although I'm not sure if you really *need* to address drift.

But if you wanted to tackle drift, you really need something that constantly decelerates the ship. This is not unlike the afterburner solution. The idea is that you're constantly exerting force against acceleration. It shouldn't be particularly powerful since you're constantly decelerating -- which means you would accelerate slower and need to exert more thrust to maintain top speed (possibly even including turning). The nature of this outfit implies reverse (weapon space and slows forward acceleration) and side thrusters (outfit space and slows turning), but not engine space, since ships without engines won't fly. The real selling point of this over the afterburner solution is that it is constant and therefore completely compatible with afterburners, but you do give up some acceleration.

TL;DR:
TO Stop:
Adjust reverse thrusters so you can't drive backwards either by outfit or setting.

Accounting for drift:
An outfit that causes you to constantly decelerate would cause your ship to stop naturally.

Sadly both of these probably require a small amount of code rather than simple outfits. But neither require key presses to use.

**Comment by marphod** [comment]
We're running out of keys?  Uh?  

There are roughly 85 keys on a standard, minimal/laptop keyboard.  Even ignoring the modifier keys (shift, alt, control, system), weird ones (scroll lock, prtsc), and the f-keys, that's 60-ish. 

In the flight context, there are 24 configurable keys (in 0.9.7; i really should move to a dev build), plus the non-configurable ones (that i know of, there is zoom in/out, and something with the number keys to group ships).  That leaves around 20 unused, by my count.  

Beyond that, just because a command can be mapped to a key doesn't mean there has to be an active key mapping for it.  

(and trying to limit UI decisions so controls are trivially mapped between a controller and keyboard is a bad idea.  There are differences in UX and expectations, and trying for identical UIs leads to boneheaded decisions, like ME:A limiting the user to 3 active abilities on all platforms)

---

Really, I'd be happy with shift-Reverse actively stopping my ship.


---

Reasons: 

Because it feels like an instinctual control that's missing.

Because I mostly play one-handed as the other is supporting my laptop, so pointer-based UI (e.g. selecting ships) takes me longer and is annoyingly hard when i keep drifting around.

Because my laptop's GPU can get easily overloaded, and having combat masses being offscreen eases the load, but I want my flagship nearby so i have a chance at boarding before targets get destroyed in crossfire.  The concussive drifting plus makes that hard.

**Comment by endless-sky** [comment]
The issue with adding new key shortcuts for things that only a tiny number of people will use is that it becomes hard, especially for new players, to spot which controls in the preferences are the actual important ones that they need to remember.

There are no situations in the game where it's necessary to come to a complete stop instead of a near-complete stop, unless you're counting someone who wants to be able to leave the game running for an hour without their ship moving. (And, that's far enough outside the intended use of the game that there's no reason to add controls just to support it.)

**Comment by Musaran2** [comment]
Make it possible to select flag ship, and then use escort command to hold position or go to location ?

There would surely be also other uses for selecting yourself.
